### DISCUSSION TITLE
Handling 8khz Audio files for speech to text

### DISCUSSION THREAD

**Discussion by utkarshpandey12** [speechly/speechly#192] [Q&A] [answered]
Hi,
I will try to make the question as detailed as possible.
So I have managed to run speechly using grpc API's. **Issue is that sampling rate of audio for speechly services is fixed at 16khz. My audio files are call recordings generated by twilio at 8khz which is standard as per twilio docs**. [https://support.twilio.com/hc/en-us/articles/223180588-Best-Practices-for-Audio-Recordings](url). 
**I checked methods to upscale the sampling rate from 8khz to 16khz but I see a divided opinion on this according to google speech to text docs and few other online sources.
So on using speechly grpc api on these 8khz audio recordings speechly is returning empty string.**
**Here's the 8khz audio file** - [https://api.twilio.com/2010-04-01/Accounts/AC54397df1de0bfff6814a69953159d2cc/Recordings/RE12ab4cbdfa1f3192fd761756a1973549](url).

**please refer to the images to see about google documentation about upsampling and another online source using a sox client to do so.
I am confused because without upsampling speechly is returning nothing but according to online sources its irrelevant to do so**

**How to handle these 8khz audio files for speech to text purposes.**

![Screenshot_2022-05-19-09-32-19-39_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/38834517/169237614-bb476c41-b636-48cb-b7f9-2f06ed4a60d6.jpg)
![Screenshot_2022-05-19-09-33-38-12_40deb401b9ffe8e1df2f1cc5ba480b12](https://user-images.githubusercontent.com/38834517/169237578-1891c5e1-ffd2-45f5-999d-f1101e58076b.jpg)

**Comment by bigdatabaracus** [comment, marked as answer]
Hello @utkarshpandey12, thank you for reaching out. A great question! The Speechly ASR system has been trained using 16 kHz audio. You may upsample your 8 kHz audio to 16 kHz put unfortunately the upsample process does not recover the frequency bands that have been lost when the audio has been stored to 8 kHz format. As the 16 kHz model _does_ utilize frequency bands that are missing from the 8 kHz format, even if you upsampled the audio, the resutsl may vary from not getting any results to getting results however with poorer accuracy. As of now Speechly can only offer 8 kHz models on the [Enterprice tier](https://www.speechly.com/pricing) in which case Speechly will train a dedicated 8 kHz model that is tailored for you use case specifically.

**Reply by utkarshpandey12** [reply]
@bigdatabaracus 
But the issue is that The telephony standard is 8-bit PCM mono uLaw with a sampling rate of 8Khz. Since this telephony format is fixed, any audio file uploaded to will be transcoded to that telephony standard. That standard being bandwidth-limited to the 300Hz - 8Khz audio range and is designed for voice and provides acceptable voice-quality results.
So unless and until the standard changes pretty much every telephony vendor is going to provide call recordings in the same format.

**Reply by bigdatabaracus** [reply]
@utkarshpandey12 You are right. Unfortunately, telephony generated audio is not a prominent use case among the users or Speechly's free and professional tiers. Use cases, where audio originates from web and mobile app, where typically the sample rate is 16 kHz or higher, are the prominent ones. This is the reason why Speechly supports 8 kHz only on the enterprise tier.
